src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,40 +1,66 @@
 import React from 'react';
 
+import { useAppSelector } from '../../app/hooks';
+import { useDispatch } from 'react-redux';
+import { filterSlice } from '../../features/filter';
+import { Status } from '../../types/Status';
+
 export const TodoFilter: React.FC = () => {
+  const filter = useAppSelector(state => state.filter);
+  const query = filter.query;
+  const dispatch = useDispatch();
+
+  const handleSelect = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    dispatch(filterSlice.actions.setStatus(event.target.value));
+  };
+
+  const ChangeQuery = (event: React.ChangeEvent<HTMLInputElement>) => {
+    dispatch(filterSlice.actions.setQuery(event.target.value));
```

Comment:
Similar to the previous comment, the `event.target.value` is being dispatched directly without any checks. While this might not be a problem in this case, it's generally a good practice to validate or sanitize user input before using it.